File: server/app/routers/catalog.py
"""GET /api/catalog/* —— HyperFrames catalog 元数据查询。

返回的是 services/catalog/data/catalog.json 的 snapshot。前端 PackagingPanel
/ FrameDesignPicker / CatalogPicker 用它列出可选 transition / caption / cover
block 缩略图。

不带认证；纯读；snapshot 随仓库 ship。
"""
from __future__ import annotations

from typing import Optional

from fastapi import APIRouter, Query
from pydantic import BaseModel, Field

from ..services.catalog import load_catalog, list_items, find_by_name, CatalogCategory

router = APIRouter(prefix="/api/catalog", tags=["catalog"])


class CatalogItem(BaseModel):
    name: str
    title: Optional[str] = None
    description: str = ""
    tags: list[str] = Field(default_factory=list)
    kind: str  # block | component
    category: str  # transition | caption | vfx | overlay | data-viz | cover | code-snippet | other
    duration: Optional[float] = None
    preview_video: Optional[str] = None
    preview_poster: Optional[str] = None


class CatalogListResponse(BaseModel):
    source: str
    version: str
    license: str = "Apache-2.0"
    items: list[CatalogItem]
    total: int


@router.get("/blocks", response_model=CatalogListResponse)
def list_catalog(
    category: Optional[str] = Query(None, description="过滤分类，例如 transition / caption / cover / vfx / overlay / data-viz"),
    tag: Optional[str] = Query(None, description="过滤 tag（单值，OR 匹配）"),
    limit: Optional[int] = Query(None, ge=1, le=200),
) -> CatalogListResponse:
    catalog = load_catalog()
    items = list_items(
        category=category,  # type: ignore[arg-type]
        tags=[tag] if tag else None,
        limit=limit,
    )
    return CatalogListResponse(
        source=catalog.get("source", "heygen-com/hyperframes"),
        version=catalog.get("version", "snapshot"),
        license=catalog.get("license", "Apache-2.0"),
        items=[CatalogItem(**i) for i in items],
        total=len(items),
    )


@router.get("/blocks/{name}", response_model=CatalogItem)
def get_catalog_item(name: str) -> CatalogItem:
    item = find_by_name(name)
    if item is None:
        from fastapi import HTTPException
        raise HTTPException(status_code=404, detail=f"catalog item '{name}' not found")
    return CatalogItem(**item)

File: Seecript-main/web/src/api/catalog.ts
/**
 * /api/catalog —— HyperFrames catalog 元数据查询。
 *
 * 列出 + 单查 HyperFrames 仓库的 block/component 元信息，给 FrameDesignPicker
 * 和 PackagingPanel 的 CatalogPicker 渲染缩略图与名字。所有静态资源
 * (preview_video / preview_poster) 都直链 HeyGen 静态站点，没有跨域和签名问题。
 */
import { api } from './client'
import type { CatalogCategory, CatalogItem, CatalogListResponse } from '../types/schemas'

export interface ListCatalogParams {
  category?: CatalogCategory
  tag?: string
  limit?: number
}

export async function listCatalog(params: ListCatalogParams = {}): Promise<CatalogListResponse> {
  const search = new URLSearchParams()
  if (params.category) search.set('category', params.category)
  if (params.tag) search.set('tag', params.tag)
  if (params.limit != null) search.set('limit', String(params.limit))
  const qs = search.toString()
  const path = qs ? `/api/catalog/blocks?${qs}` : '/api/catalog/blocks'
  return api.get<CatalogListResponse>(path)
}

export async function getCatalogItem(name: string): Promise<CatalogItem> {
  return api.get<CatalogItem>(`/api/catalog/blocks/${encodeURIComponent(name)}`)
}
